app/controllers/concerns/recents.rb

```diff
@@ -2,10 +2,24 @@ module Recents
   def recents
     ps = params.dup
     ps.delete "#{resource_name}_id"
-    render json_api: RecentSerializer.page(
-      ps,
-      Recent.where(:"#{resource_name}_id" => resource_ids),
-      { url_prefix: "#{resource_sym.to_s.pluralize}/#{resource_ids}" }
-    )
+    read_recents_from_database do
+      render json_api: RecentSerializer.page(
+        ps,
+        Recent.where(:"#{resource_name}_id" => resource_ids),
+        { url_prefix: "#{resource_sym.to_s.pluralize}/#{resource_ids}" }
+      )
+    end
+  end
+
+  private
+
+  def read_recents_from_database
```

Comment:
Wonder if we should just turn this into a globally available `Panoptes.read_from_replica(flipper_key)` method? 2/10 tho

Reply:
thought about it as well, do we want granularity to have the dumps run on the replica but switch this feature back (say the replica lag is really high)?

Reply:
Maybe? That's why I was passing in the flipper key, but I'm happy to leave it off if you think we don't need it.

Reply:
ha! bad cam, I read that completely differently. i'll extract the common code block to something like that.